web/source/text/deadkeys.ts

```diff
@@ -57,7 +64,13 @@ namespace com.keyman.text {
 
     clone(): DeadkeyTracker {
       let dkt = new DeadkeyTracker();
-      dkt.dks = this.toSortedArray();
+      let dks = this.toSortedArray();
+
+      // Make sure to clone the deadkeys themselves - the Deadkey object is mutable.
+      dkt.dks = [];
```

Comment:
Could you use `const copied = { ...original }` (object spread operator) to simplify cloning?